service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/controller/utils/ReportItkHeaderParserUtil.java

```diff
@@ -23,8 +24,22 @@ public ItkReportHeader getHeaderValues(Element headerElement) {
 
     private void setAddressList(ItkReportHeader header, Element headerElement) {
         List<Element> elements = getElements(headerElement, ITK_ADDRESS_NODE);
-        List<String> addresses = elements.stream().map(it -> it.attributeValue("uri")).collect(toList());
-        header.setAddressList(addresses);
+        Optional<String> dosServiceId = elements.stream()
+            .filter(it -> it.attribute("type") != null)
```

Comment:
could we check if type is equal to "2.16.840.1.113883.2.1.3.2.4.18.44"?

Reply:
Done.